src/main/java/edu/ucla/library/iiif/auth/services/AccessCookieCryptoServiceImpl.java

```diff
@@ -134,12 +162,20 @@
 
     @Override
     public Future<JsonObject> decryptCookie(final String aCookieValue) {
-        final JsonObject decodedCookie = new JsonObject(new String(Base64.getDecoder().decode(aCookieValue)));
-        final byte[] encryptedCookieData = decodedCookie.getBinary(CookieJsonKeys.SECRET);
-        final byte[] nonce = decodedCookie.getBinary(CookieJsonKeys.NONCE);
+        final JsonObject decodedCookie;
+        final byte[] encryptedCookieData;
+        final byte[] nonce;
         final byte[] serializedCookieData;
         final JsonObject cookieData;
 
+        try {
+            decodedCookie = new JsonObject(new String(Base64.getDecoder().decode(aCookieValue)));
```

Comment:
Could this `decodedCookie` variable be defined in the try scope? If it's not used outside of it, it seems cleaner to keep it isolated to that scope.